### PR TITLE
nixos/cde: add extraPackages option

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/desktop-managers/cde.nix [hunk 1]
```diff
@@ -8,9 +8,26 @@ let
 in {
   options.services.xserver.desktopManager.cde = {
     enable = mkEnableOption "Common Desktop Environment";
+
+    extraPackages = mkOption {
```
Comment (markuskowa) [inline nixos/modules/services/x11/desktop-managers/cde.nix:12]: What is the motivation to have an extra option here? Could you not just use `environment.systemPackages` directly?

Comment (gnidorah) [reply nixos/modules/services/x11/desktop-managers/cde.nix:12]: @markuskowa Its more nicer to put CDE specific packages there then to put them under systemPackages. mate, xmonad, i3, exwm have this option too.

File: nixos/modules/services/x11/desktop-managers/cde.nix [hunk 2]
```diff
@@ -8,9 +8,26 @@ let
...
+      type = with types; listOf package;
+      default = with pkgs; with xorg; [
+        xclock bitmap xlsfonts xfd xrefresh xload xwininfo xdpyinfo xwd xwud
+      ];
+      example = literalExample ''
+        with pkgs; with xorg; [
```
Comment (markuskowa) [inline nixos/modules/services/x11/desktop-managers/cde.nix:18]:
```suggestion
        with pkgs.xorg; [
```

Comment (gnidorah) [reply nixos/modules/services/x11/desktop-managers/cde.nix]: @markuskowa Applied, thanks!

**Comment by gnidorah** [comment]
@markuskowa Thank you!

**Pull request merged**